### ISSUE TITLE
Failure to start Slicer from Linux package built against VTK 9.1

### ISSUE THREAD

**Issue by jcfr** [Slicer/Slicer#6231]
Following integration of https://github.com/Slicer/Slicer/pull/5978 addressing https://github.com/Slicer/Slicer/issues/5956, attempt to start the Slicer fails with the following message:

```
/home/jcfr/Downloads/Slicer-4.13.0-2022-02-25-linux-amd64/bin/SlicerApp-real: error while loading shared libraries: libOpenGL.so.0: cannot open shared object file: No such file or directory
```

## Steps to reproduce

| Works | Package name | Link | 
|--|--|--|
| :heavy_check_mark:  | `Slicer-4.13.0-2022-02-24-linux-amd64.tar.gz` | https://download.slicer.org/?revision=30644 |
| :x:  | `Slicer-4.13.0-2022-02-25-linux-amd64.tar.gz` | https://download.slicer.org/?revision=30654 |
| :x:  | `Slicer-4.13.0-2022-02-28-linux-amd64.tar.gz` | https://download.slicer.org/?revision=30663 |

## Proposed solution

Backports the following commits referenced in https://github.com/Slicer/Slicer/issues/5956:

| Slicer/VTK commit | Description | Related VTK issues or merge request | Discussion |
|--|--|--|--|
| https://github.com/Slicer/VTK/commit/60b13991d37445f70a8e24604a45047022e6cb18 | `COMP: Remove custom FindOpenGL to ensure OpenGL_GL_PREFERENCE is considered` |  [VTK#17925](https://gitlab.kitware.com/vtk/vtk/-/issues/17925) | https://discourse.cmake.org/t/adding-gles-support-to-findopengl/2087 |
| https://github.com/Slicer/VTK/commit/e88e47c9679cb897243abefa05bc391866e70306 | `COMP: Ensure project depending on VTK link against correct OpenGL libraries` | [VTK#7911](https://gitlab.kitware.com/vtk/vtk/-/merge_requests/7911), [VTK#7013](https://gitlab.kitware.com/vtk/vtk/-/merge_requests/7013) | NA |

Since merge requests [VTK#7911](https://gitlab.kitware.com/vtk/vtk/-/merge_requests/7911) and [VTK#7013](https://gitlab.kitware.com/vtk/vtk/-/merge_requests/7013) have already been integrated, it should be a matter of backporting https://github.com/Slicer/VTK/commit/60b13991d37445f70a8e24604a45047022e6cb18 

## Expected behavior

Slicer starts.

## Environment
- Slicer version: See above
- Operating system: Linux

**Comment by jcfr** [comment]
This is issue is similar to https://github.com/Slicer/Slicer/issues/5228

**Comment by jcfr** [comment]
It turns out the fixes integrated in VTK (see [VTK#7911](https://gitlab.kitware.com/vtk/vtk/-/merge_requests/7911) & [VTK#7013](https://gitlab.kitware.com/vtk/vtk/-/merge_requests/7013) ) where not functionally equivalent to the one we had in our `Slicer/VTK` fork as [Slicer/VTK@e88e47c](https://github.com/Slicer/VTK/commit/e88e47c9679cb897243abefa05bc391866e70306)

It turns out that (1) [VTK#7911](https://gitlab.kitware.com/vtk/vtk/-/merge_requests/7911) was adding to `vtk-config` the variable `OPENGL_GL_PREFERENCE` instead of `OpenGL_GL_PREFERENCE` and (2) the exported preference was always `GLVND` on system where both the `LEGACY` and `GLVND` libraries were available.

This was preventing the lookup of the correct OpenGL libraries in project built against VTK.

A fix has been submitted upstream as https://gitlab.kitware.com/vtk/vtk/-/merge_requests/8945 and backported in our `Slicer/VTK` fork.

For reference:

> COMP: Ensure project depending on VTK link against correct OpenGL libraries 
> 
> This commit is a follow-up of [ab02c350](https://github.com/vtk/vtk/-/commit/ab02c3506cb1b61282382a2264b242766e21b346) (vtk-config: export OPENGL_GL_PREFERENCE to the package) introduced by [!7911 (merged)](https://gitlab.kitware.com/vtk/vtk/-/merge_requests/7911) where (1) the variable was improperly cased and (2) the
> opengl_preference was systematically set to GLVND even if OpenGL::GL was
> defined.
> 
> This commit implements two key changes:
> 
> * updates the variable name to match the name expected by the
>   FindOpenGL.cmake module and ensures project depending on VTK will
>   look-up the expected OpenGL libraries.
> 
> * sets opengl_preference by also explicitly testing the variable "use_legacy"
>   Since on some system both set of libraries can be found (GLVND and LEGACY),
>   it is critical to explicitly check the user choice.
> 
> This is required to successfully configure project like CTK, ITK or VTK
> module wheels and ensure the corresponding libraries are not linked
> against libOpenGL when OpenGL_GL_PREFERENCE is set to LEGACY.

_Originally posted by @jcfr in https://github.com/Slicer/Slicer/issues/6232#issuecomment-1055128967_

**Comment by jcfr** [comment]
Downloading the latest nightly allow to confirm the fix works as expected.
